### PR TITLE
Translate faq functions page

### PR DESCRIPTION
**Pull request by orrsteinberg**
Translated the 'Passing Functions to Components' page in the FAQ section, as mentioned in https://github.com/reactjs/he.reactjs.org/issues/1#issuecomment-767143583

Let me know if any changes are needed. 🙂

### CODE REVIEW

File: content/docs/faq-functions.md [hunk 1]
```diff
@@ -217,23 +217,23 @@ class Alphabet extends React.Component {
...
-> Note:
+> הערה:
 >
-> `_.debounce`, `_.throttle` and `raf-schd` provide a `cancel` method to cancel delayed callbacks. You should either call this method from `componentWillUnmount` _or_ check to ensure that the component is still mounted within the delayed function.
+> בכדי לבטל callbacks מעוכבים, `_.debounce`, `_.throttle` ו-`raf-schd` מספקים מתודת `cancel`. עליך לקרוא למתודה הזאת מתוך `componentWillUnmount` _או_ לוודא מתוך הפונקציה המעוכבת שהקומפוננטה עדיין מעוגנת (mounted).
 
 #### Throttle {#throttle}
 
-Throttling prevents a function from being called more than once in a given window of time. The example below throttles a "click" handler to prevent calling it more than once per second.
+'Throttling' מונע מפונקציה להיקרא יותר מפעם אחת בתוך חלון זמן נתון. הדוגמה למטה עושה throttle למטפל אירוע לחיצה כדי למנוע מקריאה לפונקציה יותר מפעם אחת בשנייה.
```
Comment (galtalmor) [inline content/docs/faq-functions.md:236]:
```suggestion
Throttling מונע מפונקציה להיקרא יותר מפעם אחת בתוך חלון זמן נתון. הדוגמה למטה עושה throttle למטפל אירוע לחיצה כדי למנוע מקריאה לפונקציה יותר מפעם אחת בשנייה.
```

File: content/docs/faq-functions.md [hunk 2]
```diff
@@ -261,7 +261,7 @@ class LoadMoreButton extends React.Component {
 
 #### Debounce {#debounce}
 
-Debouncing ensures that a function will not be executed until after a certain amount of time has passed since it was last called. This can be useful when you have to perform some expensive calculation in response to an event that might dispatch rapidly (eg scroll or keyboard events). The example below debounces text input with a 250ms delay.
+'Debouncing' מוודא שהפונקציה לא תוצא לפועל עד שפרק זמן מסוים חלף מאז הפעם האחרונה בה היא נקראה. זה יכול להיות שימושי כאשר אתה צריך לבצע חישוב יקר כלשהו בתגובה לאירוע שעלול להיות משוגר במהירות (למשל אירועי גלילה או הקלדה). הדוגמה למטה עושה debounce לקלט טקסט עם דיליי של 250ms.
```
Comment (galtalmor) [inline content/docs/faq-functions.md:264]:
```suggestion
Debouncing מוודא שהפונקציה לא תוצא לפועל עד שפרק זמן מסוים חלף מאז הפעם האחרונה בה היא נקראה. זה יכול להיות שימושי כאשר אתה צריך לבצע חישוב יקר כלשהו בתגובה לאירוע שעלול להיות משוגר בתדירות גבוהה (למשל אירועי גלילה או הקלדה). הדוגמה למטה עושה debounce לקלט טקסט עם השהייה של 250 מילי שניות.
```

File: content/docs/faq-functions.md [hunk 3]
```diff
@@ -299,12 +299,11 @@ class Searchbox extends React.Component {
 ```
 
 #### `requestAnimationFrame` throttling {#requestanimationframe-throttling}
+[`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) היא דרך לקביעת תור להוצאה לפועל של פונקציה בתוך הדפדפן בזמן האופטימלי מבחינת ביצועי רינדור. פונקציה שממתינה בתור עם `requestAnimationFrame` תשוגר בפריים הבא. הדפדפן יעבוד קשה בכדי לוודא שישנם 60 פריימים לשנייה (60 fps). יחד עם זאת, אם הדפדפן לא מסוגל לעשות זאת, הוא *יגביל* באופן טבעי את כמות הפריימים בשנייה. לדוגמה, מכשיר מסוים עלול להיות מסוגל להתמודד רק עם 30 פריימים לשנייה כך שתקבל רק 30 פריימים באותה שנייה. שימוש ב-`requestAnimationFrame` בשביל לבצע throttling הוא טכניקה שימושית בכך שזה מונע ממך לבצע יותר מ60 עידכונים בשנייה. אם אתה מבצע 100 עידכונים בשנייה זה יוצר עבודה נוספת לדפדפן שהמשתמש בכל מקרה לא ייראה.
```
Comment (galtalmor) [inline content/docs/faq-functions.md:302]:
```suggestion

[`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) היא דרך לקביעת תור להוצאה לפועל של פונקציה בתוך הדפדפן בזמן האופטימלי מבחינת ביצועי רינדור. פונקציה שהוכנסה לתור עם `requestAnimationFrame` תבוצע בפריים הבא. הדפדפן יעבוד קשה בכדי לוודא שישנם 60 פריימים לשנייה (60 fps). יחד עם זאת, אם הדפדפן לא מסוגל לעשות זאת, הוא *יגביל* באופן טבעי את כמות הפריימים בשנייה. לדוגמה, מכשיר מסוים עלול להיות מסוגל להתמודד רק עם 30 פריימים לשנייה כך שתקבל רק 30 פריימים באותה שנייה. שימוש ב-`requestAnimationFrame` בשביל לבצע throttling היא טכניקה שימושית בכך שהיא מונעת ממך לבצע יותר מ-60 עידכונים בשנייה. אם אתה מבצע 100 עידכונים בשנייה זה יוצר עבודה נוספת לדפדפן שהמשתמש בכל מקרה לא ייראה.
```

File: content/docs/faq-functions.md [hunk 4]
```diff
@@ -299,12 +299,11 @@ class Searchbox extends React.Component {
...
 
-[`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) is a way of queuing a function to be executed in the browser at the optimal time for rendering performance. A function that is queued with `requestAnimationFrame` will fire in the next frame. The browser will work hard to ensure that there are 60 frames per second (60 fps). However, if the browser is unable to it will naturally *limit* the amount of frames in a second. For example, a device might only be able to handle 30 fps and so you will only get 30 frames in that second. Using `requestAnimationFrame` for throttling is a useful technique in that it prevents you from doing more than 60 updates in a second. If you are doing 100 updates in a second this creates additional work for the browser that the user will not see anyway.
-
->**Note:**
+> הערה:
```
Comment (galtalmor) [inline content/docs/faq-functions.md:304]:
```suggestion

> הערה:
```

File: content/docs/faq-functions.md [hunk 5]
```diff
@@ -315,20 +314,20 @@ class ScrollListener extends React.Component {
...
     this.scheduleUpdate = rafSchedule(
       point => this.props.onScroll(point)
     );
   }
 
   handleScroll(e) {
-    // When we receive a scroll event, schedule an update.
-    // If we receive many updates within a frame, we'll only publish the latest value.
+    // כאשר אנחנו מקבלים אירוע גלילה, קבע עידכון.
+    // אם אנחנו מקבלים עידכונים רבים בפרק זמן קצוב, נפרסם רק את הערך העדכני ביותר.
```
Comment (galtalmor) [inline content/docs/faq-functions.md:325]:
```suggestion
    // אם אנחנו מקבלים עידכונים רבים במשך פריים אחד, נפרסם רק את הערך האחרון.
```

**Review by galtalmor — CHANGES_REQUESTED** [review]

**Comment by orrsteinberg** [comment]
Great, thank you for taking the time to review the whole thing :+1: 
Changes have been applied.

**Review by galtalmor — APPROVED** [review]

**Comment by galtalmor** [comment]
@orrsteinberg Thank you for the contribution!

**Pull request merged**